drivers/mpu9250/MPU9250.cpp

```diff
@@ -414,90 +413,93 @@ void MPU9250::_measure()
 	// Get FIFO byte count to read and floor it to the report size.
 	int bytes_to_read = get_fifo_count() / sizeof(int_status_report) * sizeof(int_status_report);
 
-	const unsigned buf_len = 52 * sizeof(int_status_report);
+	// The FIFO buffer on the MPU is 512 bytes according to the datasheet, so let's use
+	// 36*14 = 504.
+	const unsigned buf_len = 36 * sizeof(int_status_report);
```

Comment:
would be nicer to define `FIFO_LEN` (or similar) and calculate `buf_len` based on the maximum number of reports that fit.